src/main/java/htmlpublisher/workflow/WorkflowActionsFactory.java

```diff
@@ -56,22 +56,27 @@ public Collection<? extends Action> createFor(Job j) {
         List<Action> actions = new LinkedList<>();
         if (j.getClass().getCanonicalName().startsWith("org.jenkinsci.plugins.workflow"))
         {
-            final Run<?,?> r = j.getLastSuccessfulBuild();
-            if (r != null) {
-                // If reports are being saved on the build level (keep for all builds)
-                List<HtmlPublisherTarget.HTMLBuildAction> reports = r.getActions(HtmlPublisherTarget.HTMLBuildAction.class);
+            final Run<?,?> lastSuccessful = j.getLastSuccessfulBuild();
+            final Run<?,?> lastBuild = j.getLastBuild();
```

Comment:
Are you sure you want this and not `lastCompleteBuild`?
`lastBuild` may still be running so reports may not be finalized.

Reply:
Oh I was not aware of a broken link. If true, that is clearly a bug.